src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -451,6 +451,22 @@
       <summary>Creates a PostfixUnaryExpressionSyntax node.</summary>
     </FactoryComment>
   </Node>
+  <Node Name="WithExpressionSyntax" Base="ExpressionSyntax">
+    <!-- PROTOTYPE: Add comments for all of these things -->
+    <Kind Name="WithExpression"/>
+    <Field Name="Receiver" Type="ExpressionSyntax" />
+    <Field Name="WithKeyword" Type="SyntaxToken">
+      <Kind Name="WithKeyword" />
+    </Field>
+    <Field Name="OpenBraceToken" Type="SyntaxToken">
+      <Kind Name="OpenBraceToken"/>
+    </Field>
+    <!-- PROTOTYPE: This is the same syntax as for an anonymous object creation, but do we want to share the nodes? -->
+    <Field Name="Initializers" Type="SeparatedSyntaxList&lt;AnonymousObjectMemberDeclaratorSyntax&gt;" AllowTrailingSeparator="true" />
```

Comment:
Have you considered nested withers? Something like:

`record with { Owner = { Name = "name" } }` -> 
`record.With(owner: record.Owner.With(name: "name"))`

Also, it seems like collections could use some help, we already have this pattern:

`node with { ArgList = { Args = { 1 } }` -> 
`node.With(argList: node.ArgList.With(args: node.ArgList.Args.Add(1))`

though with that syntax it's not clear if we want `Add` or a complete replacement.

So I think it would be actually more similar to object/collection initializers.

Reply:
Nested withers is a neat idea, but out of scope for this PR, I think. We would have to discuss it and approve it in the design meeting.

Reply:
Filed https://github.com/dotnet/csharplang/issues/3262